### PR TITLE
COTR-19 Implemented defend melee functionality

### PR DESCRIPTION
**Pull request by Elijuszek**

### CODE REVIEW

File: Scenes/Actors/Player/Player.cs [hunk 1]
```diff
@@ -40,6 +40,7 @@ public partial class Player : Actor
 
     private Globals globals; // Object which handel Global actions (Saving, Loading)
     public WeaponsManager WeaponsManager { get; set; } // For handeling weapons
+    private Timer defendTimer;
```
Comment (Karolinskis) [inline Scenes/Actors/Player/Player.cs:43]: Comment this variable like the ones before it

File: Scenes/Actors/Player/Player.cs [hunk 2]
```diff
@@ -338,6 +343,16 @@ private void AttackTimerTimeout()
         //    Deliver();
         //}
     }
+
+    /// <summary>
+    /// Method for handling defend timer timout by setting defualt values
```
Comment (Karolinskis) [inline Scenes/Actors/Player/Player.cs:348]: Typo. Should be default

**Review by Karolinskis — CHANGES_REQUESTED** [review]

File: Scenes/Actors/Player/Player.cs [hunk 3]
```diff
@@ -18,6 +18,7 @@ public partial class Player : Actor
     //TODO: lacking Joystick scene implementation
     //private Joystick movementJoystick;
     //private Joystick attackJoystick;
+    public WeaponsManager WeaponsManager { get; set; } // For handeling weapons
```
Comment (Karolinskis) [inline Scenes/Actors/Player/Player.cs:21]: Public variables should have summaries for example:
```csharp
/// <summary>
/// something
/// </summary>
public int variable;
````

File: Scenes/Actors/Player/Player.cs [hunk 4]
```diff
@@ -18,6 +18,7 @@ public partial class Player : Actor
...
 
     private AnimationPlayer animationPlayer;
```
Comment (Karolinskis) [inline Scenes/Actors/Player/Player.cs:23]: Missing comments

File: Scenes/Actors/Player/Player.cs [hunk 5]
```diff
@@ -18,6 +18,7 @@ public partial class Player : Actor
...
     private RemoteTransform2D cameraTransform;
```
Comment (Karolinskis) [inline Scenes/Actors/Player/Player.cs:24]: Missing comments

File: Scenes/Actors/Player/Player.cs [hunk 6]
```diff
@@ -28,7 +29,6 @@ public partial class Player : Actor
 
     // Level system, whcih handels obtained xp, levelUp and obtaining skills
     private LevelSystem levelSystem;
-
     private PackedScene bloodScene;
     private PackedScene damagePopup;
```
Comment (Karolinskis) [inline Scenes/Actors/Player/Player.cs:33]: Missing comments

File: Scenes/Actors/Player/Player.cs [hunk 7]
```diff
@@ -39,7 +39,9 @@ public partial class Player : Actor
     //private GUI gui;
```
Comment (Karolinskis) [inline Scenes/Actors/Player/Player.cs:39]: If it's unneeded, delete it

**Review by Karolinskis — CHANGES_REQUESTED** [review]

File: Scenes/Actors/Player/Player.cs (hunk 1 above)
Comment (Elijuszek) [reply Scenes/Actors/Player/Player.cs]: Done

File: Scenes/Actors/Player/Player.cs (hunk 2 above)
Comment (Elijuszek) [reply Scenes/Actors/Player/Player.cs]: Done

File: Scenes/Actors/Player/Player.cs (hunk 3 above)
Comment (Elijuszek) [reply Scenes/Actors/Player/Player.cs:21]: Done

File: Scenes/Actors/Player/Player.cs (hunk 4 above)
Comment (Elijuszek) [reply Scenes/Actors/Player/Player.cs:23]: Done

File: Scenes/Actors/Player/Player.cs (hunk 5 above)
Comment (Elijuszek) [reply Scenes/Actors/Player/Player.cs:24]: Done

File: Scenes/Actors/Player/Player.cs (hunk 6 above)
Comment (Elijuszek) [reply Scenes/Actors/Player/Player.cs:33]: Done

File: Scenes/Actors/Player/Player.cs (hunk 7 above)
Comment (Elijuszek) [reply Scenes/Actors/Player/Player.cs:39]: Done

**Review by Karolinskis — APPROVED** [review]

**Pull request merged**